lib/types/subdocument.js

```diff
@@ -229,12 +229,12 @@ Subdocument.prototype.$ignore = function(path) {
  */
 
 Subdocument.prototype.ownerDocument = function() {
+  this.isSubDocument = true;
```

Comment:
`this.$__.isSubDocument = true`

Also, put this line in the constructor, not in `ownerDocument()`